test/fuzz/luaL_loadbuffer/serializer.cc

```diff
@@ -931,7 +943,11 @@ NESTED_PROTO_TOSTRING(IndexWithExpression, indexexpr, Variable)
 {
 	std::string indexexpr_str = PrefixExpressionToString(
 		indexexpr.prefixexp());
-	indexexpr_str += "[" + ExpressionToString(indexexpr.exp()) + "]";
+	indexexpr_str += "[";
+	/* Prevent errors 'table index is nil' and 'table index is NaN'. */
+	indexexpr_str += AllowedIndexExpressionToString(indexexpr.exp());
```

Comment:
I suppose that this part is excess.
For example, the following code is OK for Lua and LuaJIT:
```
lua -e '
local t = {}
local a = 1
t[a] = t[nil]
'
```
The table index NaN/nil is prohibited only for storing operations.

Reply:
Reverted in `IndexWithExpression` and reduced a patch for this commit:

```diff
diff --git a/test/fuzz/luaL_loadbuffer/serializer.cc b/test/fuzz/luaL_loadbuffer/serializer.cc
index c6f480d1ef..b4500da9ca 100644
--- a/test/fuzz/luaL_loadbuffer/serializer.cc
+++ b/test/fuzz/luaL_loadbuffer/serializer.cc
@@ -54,9 +54,9 @@ const std::set<std::string> KReservedLuaKeywords {
 };
 
 const std::string kCounterNamePrefix = "counter_";
-const std::string kNotNaNAndNilWrapperName = "not_nan_and_nil";
 const std::string kNumberWrapperName = "always_number";
 const std::string kBinOpWrapperName = "only_numbers_cmp";
+const std::string kNotNaNAndNilWrapperName = "not_nan_and_nil";
 
 PROTO_TOSTRING(Block, block);
 PROTO_TOSTRING(Chunk, chunk);
@@ -943,11 +943,7 @@ NESTED_PROTO_TOSTRING(IndexWithExpression, indexexpr, Variable)
 {
 	std::string indexexpr_str = PrefixExpressionToString(
 		indexexpr.prefixexp());
-	indexexpr_str += "[";
-	/* Prevent errors 'table index is nil' and 'table index is NaN'. */
-	indexexpr_str += AllowedIndexExpressionToString(indexexpr.exp());
-	indexexpr_str += "]";
-
+	indexexpr_str += "[" + ExpressionToString(indexexpr.exp()) + "]";
 	return indexexpr_str;
 }
 
@@ -1107,11 +1103,9 @@ PROTO_TOSTRING(Field, field)
 
 NESTED_PROTO_TOSTRING(ExpressionAssignment, assignment, Field)
 {
-	std::string assignment_str;
-	assignment_str += "[";
 	/* Prevent error 'table index is nil' and 'table index is NaN'. */
-	assignment_str += AllowedIndexExpressionToString(assignment.key());
-	assignment_str += "]";
+	std::string assignment_str = "[ " +
+		AllowedIndexExpressionToString(assignment.key()) + " ]";
 	assignment_str += " = " + ExpressionToString(assignment.value());
 	return assignment_str;
 }
```